### PR TITLE
fix: fix old consignments nav

### PR DESCRIPTION
**Pull request by brainbicycle**
The type of this PR is: **fix**

This PR resolves [MOPLAT-132]

### Description

The old consignments upload flow used to be presented modally but was changed to be presented via push while we were building the new consignments upload flow, this means the old dismissModal nav methods no longer worked. Instead this uses popToRoot() method and shows a back button rather than an X button on the initial overview screen. 

<details><summary>Screenshots</summary>

https://user-images.githubusercontent.com/49686530/155795129-3c282368-a57e-4f1f-9641-4fff78d1f701.mp4

https://user-images.githubusercontent.com/49686530/155795146-dbe0bcd8-e954-424d-a398-25e49fca0e11.mp4

</details>

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- Fix consignments upload nav - brian 

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

[MOPLAT-132]: https://artsyproduct.atlassian.net/browse/MOPLAT-132?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (Fix consignments upload nav - brian)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against bc3600be914b902bde7255d70fe0b54d6f409f4e
</p>

**Review by MounirDhahri — APPROVED** [review]
LGTM 🙌

**Comment by MounirDhahri** [comment]
hmm, the PR is failing because of something you didn't change, this is weird

**Comment by MounirDhahri** [comment]
aha, this is happening because of mocking `popToRoot` that we send [here](https://github.com/artsy/eigen/blob/ebbc22f40657e48c52097cb9253d37075b0c6ed4/src/app/Scenes/MyCollection/MyCollection.tsx#L121)

File: src/app/Scenes/MyCollection/MyCollection.tests.tsx [hunk 1]
```diff
@@ -104,7 +104,7 @@ describe("MyCollection", () => {
 
       expect(navigate).toHaveBeenCalledWith(
         "my-collection/artworks/new",
-        expect.objectContaining({ passProps: { mode: "add" } })
+        expect.objectContaining({ passProps: { mode: "add", onSuccess: expect.anything() } })
```
Comment (MounirDhahri) [inline src/app/Scenes/MyCollection/MyCollection.tests.tsx:107]: 😄

**Review by MounirDhahri — APPROVED** [review]

File: src/app/Scenes/MyCollection/MyCollection.tests.tsx (hunk 1 above)
Comment (brainbicycle) [reply src/app/Scenes/MyCollection/MyCollection.tests.tsx:107]: 😅 I tried messing with spaces in snapshot for a bit but figured this is fine for purpose of this test

**Comment by MounirDhahri** [comment]
OMG I merged the PR instead of squashing it :(

**Pull request merged**